bluemira/base/parameter_frame/_frame.py

```diff
@@ -96,3 +97,45 @@ def from_json(cls, json_in: Union[str, json.SupportsRead]) -> NewParameterFrame:
                 return cls.from_dict(json.load(f))
         # load from a JSON string
         return cls.from_dict(json.loads(json_in))
+
+
+class ParameterSetup(abc.ABC):
```

Comment:
Can't say I'm mad about using a class here. I feel like this could be a simple `make_parameterframe(cls, params)` (or similar) function that both the builder and designer call. I'd like to avoid the case we had with the previous builders where we need to go multiple inheritance levels deep before we see where `.params` is set. 

I think having a simple factory function is more flexible, and it makes more sense as something that lives in the `parameter_frame` module.